db/migrate/20180426145633_add_stock_location_id_to_spree_shipments.spree.rb

```diff
@@ -2,5 +2,9 @@
 class AddStockLocationIdToSpreeShipments < ActiveRecord::Migration
   def change
     add_column :spree_shipments, :stock_location_id, :integer
+
+    # This Spree migration has not come with data migration and we need all shipments to have a valid stock location
+    #   This update fixes that problem
+    Spree::Shipment.update_all("stock_location_id = #{Spree::StockLocation.first.id}")
```

Comment:
Doesn't Spree migrate the data in an upcoming migration? Is a common strategy to relax constraints, migrate data and in a later migration, constraints are set back.

Reply:
I moved this topic to a separate issue here, let's talk there: #3600